### ISSUE TITLE
Hero Widget misbehavior with Text

### ISSUE THREAD

**Issue by Ahmadre** [flutter/flutter#47984]
## Steps to Reproduce

![ezgif com-resize](https://user-images.githubusercontent.com/18512224/71589164-20f58600-2b24-11ea-9994-ec3be57f67a1.gif)

1. Create a ```Text``` Widget with ```Hero``` Animation
2. Push to the Details Page where the ```Hero``` animates
3. Misbehavior like shown in the GIF above appear (__Yellow__ doubled lines appear and Font is changing!) 

**Target Platform: ALL (iOS, Android, Web)**
**Target OS version/browser: ALL**
**Devices: ALL**

## Logs

<details>
  <summary>Analyze</summary>

```
Analyzing verboto...                                                    

   info • Unused import: 'package:auto_size_text/auto_size_text.dart' •
          lib/Components/Dialogs.dart:4:8 • unused_import
   info • This class (or a class which this class inherits from) is marked as
          '@immutable', but one or more of its instance fields are not final:
          FabWithIcons.onIconTapped • lib/Components/FabList.dart:4:7 • must_be_immutable
   info • This class (or a class which this class inherits from) is marked as
          '@immutable', but one or more of its instance fields are not final:
          TakePicture.cameraController, TakePicture.heroTag •
          lib/Components/TakePicture.dart:10:7 • must_be_immutable
   info • The value of the field '_controller' isn't used • lib/Components/Tour.dart:32:23
          • unused_field
   info • The value of the local variable 'icon' isn't used •
          lib/Containers/ChatBubble.dart:39:11 • unused_local_variable
   info • The value of the local variable 'hdDelivered' isn't used •
          lib/Containers/ChatBubble.dart:40:11 • unused_local_variable
   info • This class (or a class which this class inherits from) is marked as
          '@immutable', but one or more of its instance fields are not final:
          ChatTile._onChatTap, ChatTile._memberUsername, ChatTile._herotag,
          ChatTile._memberImage, ChatTile._lastMessage, ChatTile._lastMessageTimeStamp,
          ChatTile._showLastMessageTimeStamp • lib/Containers/ChatTile.dart:5:7 •
          must_be_immutable
   info • The value of the field '_controller' isn't used •
          lib/Containers/ChatTile.dart:66:23 • unused_field
   info • Unused import: 'package:cached_network_image/cached_network_image.dart' •
          lib/Containers/NotificationsTile.dart:1:8 • unused_import
   info • Unused import: 'package:intl/intl.dart' •
          lib/Containers/NotificationsTile.dart:3:8 • unused_import
   info • This class (or a class which this class inherits from) is marked as
          '@immutable', but one or more of its instance fields are not final:
          NotificationsTile._onNotificationTap, NotificationsTile._thumbnail,
          NotificationsTile._message, NotificationsTile._notificationTimeStamp •
          lib/Containers/NotificationsTile.dart:6:7 • must_be_immutable
   info • The value of the field '_controller' isn't used •
          lib/Containers/NotificationsTile.dart:47:23 • unused_field
   info • This class (or a class which this class inherits from) is marked as
          '@immutable', but one or more of its instance fields are not final:
          PostCard._me, PostCard._postId, PostCard._onMemberTap, PostCard._onMediaTap,
          PostCard._onCommentTap, PostCard._herotag, PostCard._profileHerotag,
          PostCard._mode, PostCard._profilePic, PostCard._media, PostCard._mediaFit,
          PostCard._author, PostCard._message, PostCard._borderColor, PostCard._likes,
          PostCard._hideButtons, PostCard._isLastPost, PostCard._comments •
          lib/Containers/PostCard.dart:13:7 • must_be_immutable
   info • The value of the field '_builders' isn't used •
          lib/Helpers/VrbtnTransitionTheme.dart:9:53 • unused_field
   info • The value of the field '_animation' isn't used •
          lib/Screens/Auth/Application.dart:28:13 • unused_field
   info • Don't explicitly initialize variables to null •
          lib/Screens/Auth/Application.dart:63:12 • avoid_init_to_null
   info • This function has a return type of 'String', but doesn't end with a return
          statement • lib/Screens/Auth/Application.dart:783:44 • missing_return
   info • This function has a return type of 'String', but doesn't end with a return
          statement • lib/Screens/Auth/Application.dart:906:44 • missing_return
   info • This function has a return type of 'String', but doesn't end with a return
          statement • lib/Screens/Auth/Application.dart:1021:44 • missing_return
   info • This function has a return type of 'String', but doesn't end with a return
          statement • lib/Screens/Auth/Application.dart:1282:48 • missing_return
   info • This function has a return type of 'String', but doesn't end with a return
          statement • lib/Screens/Auth/Application.dart:1394:44 • missing_return
   info • This function has a return type of 'String', but doesn't end with a return
          statement • lib/Screens/Auth/Application.dart:1501:44 • missing_return
   info • This function has a return type of 'String', but doesn't end with a return
          statement • lib/Screens/Auth/Application.dart:1709:44 • missing_return
   info • Unused import: 'package:verboto/api/graphql.abstract.dart' •
          lib/Screens/Auth/Login.dart:18:8 • unused_import
   info • This class (or a class which this class inherits from) is marked as
          '@immutable', but one or more of its instance fields are not final: Login.email,
          Login.password • lib/Screens/Auth/Login.dart:21:7 • must_be_immutable
   info • The value of the field '_animation' isn't used •
          lib/Screens/Auth/Login.dart:36:13 • unused_field
   info • Unused import: '../../main.dart' • lib/Screens/Circles/Circles.dart:6:8 •
          unused_import
   info • Unused import: '../Messages/Messages.dart' •
          lib/Screens/Circles/Circles.dart:7:8 • unused_import
   info • Unused import: '../Events/Events.dart' • lib/Screens/Circles/Circles.dart:8:8 •
          unused_import
   info • Unused import: '../../Components/BottomBarWithFab.dart' •
          lib/Screens/Circles/Circles.dart:9:8 • unused_import
   info • The declaration '_selectedTab' isn't referenced •
          lib/Screens/Events/Events.dart:42:8 • unused_element
   info • Unused import: 'package:intl/intl.dart' • lib/Screens/Messages/Messages.dart:2:8
          • unused_import
   info • Unused import: '../Circles/Circles.dart' •
          lib/Screens/Messages/Messages.dart:11:8 • unused_import
   info • Unused import: '../Events/Events.dart' • lib/Screens/Messages/Messages.dart:12:8
          • unused_import
   info • Unused import: 'package:auto_size_text/auto_size_text.dart' •
          lib/Screens/Profile/Profile.dart:10:8 • unused_import
   info • Unused import: 'package:intl/intl.dart' • lib/Screens/Profile/Profile.dart:19:8
          • unused_import
   info • This function has a return type of 'Widget', but doesn't end with a return
          statement • lib/Screens/Profile/Profile.dart:284:18 • missing_return
   info • The declaration '_viewNetworkImage' isn't referenced •
          lib/Screens/Profile/Profile.dart:364:3 • unused_element
   info • This function has a return type of 'Widget', but doesn't end with a return
          statement • lib/Screens/Profile/Profile.dart:510:18 • missing_return
   info • The declaration '_switchPasswordVisibility' isn't referenced •
          lib/Screens/Settings/PrivacyPolicy/ScreenLock/ScreenLock.dart:167:8 •
          unused_element
   info • The declaration '_pushToBlocklist' isn't referenced •
          lib/Screens/Settings/Settings.dart:65:3 • unused_element
   info • Unused import: 'package:flutter_colorpicker/block_picker.dart' •
          lib/Screens/Stream/AddPost/AddPost.dart:5:8 • unused_import
   info • Unused import: 'package:flutter_advanced_networkimage/zoomable.dart' •
          lib/Screens/Stream/AddPost/AddPostStep2.dart:7:8 • unused_import
   info • Unused import: 'package:verboto/Helpers/CustomClipper.dart' •
          lib/Screens/Stream/AddPost/AddPostStep2.dart:11:8 • unused_import
   info • This class (or a class which this class inherits from) is marked as
          '@immutable', but one or more of its instance fields are not final:
          AddPostStep2.isPrivate • lib/Screens/Stream/AddPost/AddPostStep2.dart:20:7 •
          must_be_immutable
   info • The declaration '_viewImage' isn't referenced •
          lib/Screens/Stream/AddPost/AddPostStep2.dart:75:3 • unused_element
   info • Unused import: 'dart:io' •
          lib/Screens/Stream/AddPostIOS/AddPostIOSStep2.dart:1:8 • unused_import
   info • This class (or a class which this class inherits from) is marked as
          '@immutable', but one or more of its instance fields are not final:
          AddPostIOSStep2.originalImageInfo, AddPostIOSStep2.fileImage,
          AddPostIOSStep2.isPrivate •
          lib/Screens/Stream/AddPostIOS/AddPostIOSStep2.dart:18:7 • must_be_immutable
   info • The declaration '_viewImage' isn't referenced •
          lib/Screens/Stream/AddPostIOS/AddPostIOSStep2.dart:77:3 • unused_element
   info • Unused import: 'package:cached_network_image/cached_network_image.dart' •
          lib/Screens/Stream/index.dart:3:8 • unused_import
   info • This class (or a class which this class inherits from) is marked as
          '@immutable', but one or more of its instance fields are not final:
          Stream.onAddPost • lib/Screens/Stream/stream.dart:26:7 • must_be_immutable
   info • Unused import: 'dart:convert' • lib/Services/AppManager.dart:1:8 • unused_import
   info • Unused import: 'dart:io' • lib/Services/AppManager.dart:2:8 • unused_import
   info • Unused import: 'package:verboto/Components/Dialogs.dart' •
          lib/Services/AppManager.dart:4:8 • unused_import
   info • Unused import: 'package:flushbar/flushbar.dart' •
          lib/Services/AppManager.dart:5:8 • unused_import
   info • Unused import: 'package:flutter/services.dart' •
          lib/Services/AppManager.dart:7:8 • unused_import
   info • Unused import: 'package:flutter/widgets.dart' •
          lib/Services/local/genders/genders.service.dart:3:8 • unused_import
   info • Unused import: 'package:verboto/Services/local/i18n/VrbtnLocalizations.dart' •
          lib/Services/local/genders/genders.service.dart:4:8 • unused_import

58 issues found. (ran in 7.9s)
```

</details>

<details>
  <summary>Doctor</summary>

```
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.1 19B88, locale de-DE)
    • Flutter version 1.12.13+hotfix.5 at /Users/rebar/Documents/flutter
    • Framework revision 27321ebbad (3 weeks ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/rebar/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.8.3

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 41.1.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.41.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[✓] Connected device (1 available)
    • iPhone 11 Pro Max • 4C3E4855-3033-4F8B-BD55-ED013B547D7B • ios •
      com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)

• No issues found!
```

</details>

**Comment by Piinks** [comment]
Thanks for filing this issue!
This is a duplicate of #36220, which may be fixed by #39104.

**Comment by Ahmadre** [comment]
Thank you very much :). I searched for a duplicate, but didn't find one. I will follow up the issue there
